server/src/main/java/org/opensearch/plugins/SearchPipelinesPlugin.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.search.pipeline.Processor;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * An extension point for {@link Plugin} implementation to add custom search pipeline processors.
+ *
+ * @opensearch.api
+ */
+public interface SearchPipelinesPlugin {
```

Comment:
I was thinking about this interface as a specific alternative to `SearchPlugin`, since `SearchPlugin` offers so much power and flexibility. 

If something just implements `SearchPipelinePlugin`, it's a way of saying "this thing supplies components that modify search processing behavior at a high level in a composable way", whereas a `SearchPlugin` can do almost anything. (Of course, there's nothing preventing someone from implementing both `SearchPipelinePlugin` and `SearchPlugin`.)

@vamshin made a great analogy in conversation earlier today, comparing search pipelines to Unix pipes, which I would extend to say that processors should follow the Unix tools philosophy of doing one thing well.